emacsMacport 27.2-8.3 -> 28.1-9.0

eddsteel opened the pull request:
- https://www.gnu.org/software/emacs/news/NEWS.28.1
- https://bitbucket.org/mituharu/emacs-mac/commits/5a223c7f2ef4c31abbd46367b6ea83cd19d30aa7#chg-ChangeLog.3

###### Description of changes

Simply an update of versions and checksums, I didn't try to re-enable checks.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [X] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [X] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

eddsteel committed:
emacsMacport: 27.2-8.3 -> 28.1-9.0

- https://www.gnu.org/software/emacs/news/NEWS.28.1
- https://bitbucket.org/mituharu/emacs-mac/commits/5a223c7f2ef4c31abbd46367b6ea83cd19d30aa7#chg-ChangeLog.3

eddsteel commented on pkgs/applications/editors/emacs/macport.nix:
```diff
@@ -114,7 +113,7 @@ stdenv.mkDerivation rec {
       separately.
 
-      This is the "Mac port" addition to GNU Emacs 26. This provides a native
+      This is the "Mac port" addition to GNU Emacs. This provides a native
```
This was confusing to me, since it's not currently anything to do with Emacs 26.

isker commented:
Pinging reviewers, please take a look:
@adisbladis
@jwiegley
@matthewbauer

Thanks!

isker commented:
@adisbladis 
@jwiegley 
@matthewbauer 

I'm sorry to ping you once again, but I believe you're the only ones allowed to review and merge this and other long-outstanding emacs pull requests here in nixpkgs.  Please take a look!

alyssais commented:
@OfBorg build emacsMacport

alyssais commented:
Commit message is missing a colon, but don't worry about changing it this time unless you really want to.

alyssais commented:
OfBorg's x86_64-darwin build still produces /nix/store/1a27qgnivkvrjsjg7g01zbvl64damf1l-emacs-mac-27.2-8.3, which seems odd…

eddsteel commented:
I rebased and added a comma. Not sure what's happening with OfBorg but here's my local output:

```
nixpkgs edd$ git show -q
commit 05871e382dcc348a79be615de09b896d19e7cfef (HEAD -> emacs-macport-9.0)
Date:   Mon May 16 14:55:09 2022 -0700

    emacsMacport: 27.2-8.3 -> 28.1-9.0
    
    - https://www.gnu.org/software/emacs/news/NEWS.28.1
    - https://bitbucket.org/mituharu/emacs-mac/commits/5a223c7f2ef4c31abbd46367b6ea83cd19d30aa7#chg-ChangeLog.3
nixpkgs edd$ nix-build -A emacsMacport &>/dev/null
nixpkgs edd$ ls -l result
lrwxr-xr-x 1 edd staff 62 Jul 26 20:16 result -> /nix/store/l8yijxxxmkh2cvhr2cwhhcv1vqxlgzgy-emacs-mac-28.1-9.0
nixpkgs edd$ emacsclient -e '(emacs-version)'
"GNU Emacs 28.1 (build 1, x86_64-apple-darwin21.2.0, Carbon Version 165 AppKit 2113.2)\n of 2022-07-20"
```
